src/components/user/uc/users.uc.js

```diff
@@ -0,0 +1,67 @@
+const { ObjectId } = require('mongoose').Types;
+const { GeneralError, NotFound } = require('../../../errors');
+const { userRepo, accountRepo, trashbinRepo } = require('../repo/index');
+const restrictToSameSchool = require('../../../utils/restrictToSameSchool');
+const restrictToRole = require('../../../utils/restrictToRole');
+
+const getUserData = async (id, app) => {
+	const data = {};
+
+	const user = await userRepo.getUser(id, app);
+	if (!(user && user._id && !user.deletedAt)) {
```

Comment:
try catch around any await OR around all. getUser should throw an error it self for not found user OR return null if it is valid that they can not found a user in some cases.